pkg/sqlite/load.go

```diff
@@ -192,86 +199,101 @@ func (s *SQLLoader) AddPackageChannels(manifest registry.PackageManifest) error
 			// Get CSV for current entry
 			channelEntryCSV, err := s.getCSV(tx, channelEntryCSVName)
 			if err != nil {
-				return err
+				errs = append(errs, err)
+				break
 			}
 
 			if err := s.addProvidedAPIs(tx, channelEntryCSV, currentID); err != nil {
-				return err
+				errs = append(errs, err)
 			}
 
 			skips, err := channelEntryCSV.GetSkips()
 			if err != nil {
-				return err
+				errs = append(errs, err)
 			}
 
 			for _, skip := range skips {
 				// add dummy channel entry for the skipped version
 				skippedChannelEntry, err := addChannelEntry.Exec(c.Name, manifest.PackageName, skip, depth)
 				if err != nil {
-					return err
+					errs = append(errs, err)
+					continue
 				}
 
 				skippedID, err := skippedChannelEntry.LastInsertId()
 				if err != nil {
-					return err
+					errs = append(errs, err)
+					continue
 				}
 
 				// add another channel entry for the parent, which replaces the skipped
 				synthesizedChannelEntry, err := addChannelEntry.Exec(c.Name, manifest.PackageName, channelEntryCSVName, depth)
 				if err != nil {
-					return err
+					errs = append(errs, err)
+					continue
 				}
 
 				synthesizedID, err := synthesizedChannelEntry.LastInsertId()
 				if err != nil {
-					return err
+					errs = append(errs, err)
+					continue
 				}
 
 				if _, err = addReplaces.Exec(skippedID, synthesizedID); err != nil {
-					return err
+					errs = append(errs, err)
+					continue
 				}
 
 				if err := s.addProvidedAPIs(tx, channelEntryCSV, synthesizedID); err != nil {
-					return err
+					errs = append(errs, err)
+					continue
 				}
 
-				depth += 1
+				depth++
 			}
 
 			// create real replacement chain
 			replaces, err := channelEntryCSV.GetReplaces()
 			if err != nil {
-				return err
+				errs = append(errs, err)
+				break
 			}
 
 			if replaces == "" {
 				// we've walked the channel until there was no replacement
 				break
 			}
 
-			replaced, err := s.getCSV(tx, replaces)
+			replacedChannelEntry, err := addChannelEntry.Exec(c.Name, manifest.PackageName, replaces, depth)
```

Comment:
I would like to understand why the order of this changed?

Reply:
The change in ordering lets us add a channel entry for a "missing" replaced CSV.